### ISSUE TITLE
[Feature Request] Automate bug reporting?

### ISSUE THREAD

**Issue by anovicecodemonkey** [rg3/youtube-dl#5097]
This is a brain dump, for comment and discussion: 

I am seeing a number of poor bug reports, older versions and non-verbose output. A lot of people either don't see or don't seem to understand how to submit a proper and useful report. Would there be a way to automate this process? 

My idea is this: 
When youtube-dl fails, check the version, if it's outdated provide a link to instructions for updating and point out that a bug cannot be submitted until the software is the most recent available version, highlighting perhaps more strongly the preference for the "manual installation" method. When the version is current, create a log file and use the Github API  to partly generate or submit a report with the proper verbose output contained in the log file. 

Also -maybe- automatically check an online maintained list of common errors, for example the YouTube signature extraction bug in older versions. Detect the errors and print the advice for dealing with those errors. Things like the signature extraction bug affect versions older than X so that would be a good example of being able to provide version specific information.

**Comment by Elite** [comment]
Great idea, I would suggest using `http://pastebin.com/api` for automated creation of the log dump. The user can be asked to submit the generated link for help.

**Comment by phihag** [comment]
Yes, I have been contemplating such a change for quite some time and prepared for it step by step (for example, you can already list the current version with `-C`). Since the bug reports are not necessary public, we'd host any infrastructure on our own servers though.
